cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -271,17 +276,67 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 		return make(map[string]string), nil
 	}
 
+	parsedLabels, err := parseLabels(*kubeLabels)
+	if err != nil {
+		return nil, fmt.Errorf("instance %q had %s", instance.SelfLink, err.Error())
+	}
+	return parsedLabels, nil
+}
+
+func extractLastAppliedLabels(node *core.Node) map[string]string {
+	if lastLabels, ok := node.ObjectMeta.Annotations[lastAppliedLabelsKey]; !ok || len(lastLabels) == 0 {
+		return nil
+	} else {
+		parsedLabels, err := parseLabels(lastLabels)
+		if err != nil {
+			klog.Errorf("Failed to parse last applied labels annotation: %q, treat it as not set, err: %v", lastLabels, err)
+			return nil
+		}
+		return parsedLabels
+	}
+}
+
+func mergeManagedLabels(node *core.Node, desiredLabels map[string]string) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	lastManagedLabels := extractLastAppliedLabels(node)
```

Comment:
```suggestion
	lastAppliedLabels := extractLastAppliedLabels(node)
```
nit on variable name.